sklearn/ensemble/forest.py

```diff
@@ -72,17 +73,35 @@ class calls the ``fit`` method of each sub-estimator on random samples
 MAX_INT = np.iinfo(np.int32).max
 
 
-def _generate_sample_indices(random_state, n_samples):
+def _generate_sample_indices(random_state, n_samples, max_samples=None):
     """Private function used to _parallel_build_trees function."""
+
+    # Validate `max_samples`
+    if max_samples is None:
```

Comment:
Can't this validation occur outside of the function, keeping it lightweight and accepting only integer max_samples?

Reply:
Sure, I factored the validation and bootstrap sub-sample size generation into its own helper: https://github.com/scikit-learn/scikit-learn/pull/14682/commits/2507d5b44f278047eae2cb1c0bea4705bb266622